src/openms/include/OpenMS/FILTERING/NOISEESTIMATION/SignalToNoiseEstimator.h

```diff
@@ -105,22 +100,16 @@ namespace OpenMS
     virtual void init(const Container& c)
     {
       computeSTN_(c);
-      is_result_valid_ = true;
     }
 
     ///Return to signal/noise estimate for date point @p index
-    ///@note the first query to this function will taake longer, as
+    ///@note the first query to this function will take longer, as
     ///      all SignalToNoise values are calculated.
     ///@note you will get a warning to stderr if more than 20% of the
     ///      noise estimates used sparse windows
     virtual double getSignalToNoise(const Size index)
     {
-      if (!is_result_valid_)
-      {
-        // recompute ...
-        init(c);
-      }
-
+      OPENMS_POSTCONDITION(condition,message);
```

Comment:
this should say `OPENMS_POSTCONDITION(index < stn_estimates_.size(), "S/N estimate beyond container size was requested"); or similar